python/tchannel/zipkin/tracers.py

```diff
@@ -126,27 +133,25 @@ class TChannelZipkinTracer(object):
     buffering of any sort.
     """
 
-    def __init__(self, tchannel):
+    def __init__(self, tchannel, routers):
         """
         :param tchannel:
             A tchannel instance to send the trace info to zipkin server
+        :param routers:
+            A list contains hyperbahn instances' ip addresses
         """
         self._tchannel = tchannel
-        self.client = tchannel.request(
-            "tcollector"
-        )
+        self.client = TCollectorClient(self._tchannel)
+
+        if routers:
+            hyperbahn.advertise(self._tchannel, 'tcollector', routers)
```

Comment:
Also, the user is responsible for advertising. You don't need to do it regardless. You don't need a list of routers either. Let the TChannel pick the peer. For some level of determinism you could accept an optional hostport argument and pass that over to the TCollectorClient (it accepts an optional hostport if you want it to send requests to a specific hostport instead of letting the tchannel pick a peer).